lib/Sema/CSDiag.cpp

```diff
@@ -2559,7 +2559,7 @@ diagnoseTypeMemberOnInstanceLookup(Type baseObjTy,
       ->getAsNominalTypeOrNominalTypeExtensionContext();
   SmallString<32> typeName;
   llvm::raw_svector_ostream typeNameStream(typeName);
-  typeNameStream << nominal->getName() << ".";
+  typeNameStream << nominal->getDeclaredInterfaceType() << ".";
```

Comment:
I'm a little hesitant to print this directly. but it seems to be correct.  Could you add a test that uses a generic nested type as well?

Reply:
The declared interface type should be fine. It will use sugared generic parameter names and include the full parent type. So for a nested generic type, it should print `Outer<T>.Inner<U, V>`, etc, and we should add a test for this case.

Reply:
Actually to handle members of protocols you want getSelfInterfaceType() instead. This is identical to getDeclaredInterfaceType() for concrete types, but for protocols it has 'Self' instead of the existential.

Right now, this produces a bad fixit:
```
protocol P {}

extension P {
  static func f() {}

  func g() {
    f
  }
}

ext.swift:7:5: error: static member 'f' cannot be used on instance of type 'Self'
    f
    ^
    P.

```

Note that the correct fix is to add 'Self.', not 'P.'.

@gibachan Do you mind adding a test for this case as well?

Reply:
Thank you for kind response.
I made those fixes, but now I get a strange error on the test you advised above.
When I run the test by `utils/run-test`, the error message says "use of unresolved identifier 'f'" instead of "static member 'f' cannot be used ...". However if I run it by `../build/Ninja-ReleaseAssert/swift-macosx-x86_64/bin/swift`, it says "static member 'f' cannot be used ..." as I expect. I do not know the difference between them. Am I missing something?